extensions/cpsection/aboutcomputer/model.py

```diff
@@ -46,11 +48,38 @@ def print_aboutcomputer():
     print get_aboutcomputer()
 
 
-def get_serial_number():
+def _get_serial_number():
     serial_no = _read_device_tree('serial-number')
-    if serial_no is None:
-        serial_no = _not_available
-    return serial_no
+    if serial_no is not None:
+        return serial_no
+
+    def _helper():
+        binary = 'sugar-serial-number-helper'
+        for path in os.environ['PATH'].split(os.pathsep):
+            binary_path = os.path.join(path, binary)
+            if os.path.exists(binary_path):
+                return binary_path
+        return None
+
+    helper = _helper()
+    if helper is None:
+        return _not_available
+
+    cmd = 'pkexec %s' % helper
```

Comment:
pkexec seems to automatically resolve the full path of the executable.  Maybe it is better to let pkexec do it instead of doing the exact same thing ourselves except probably not dealing with as many edge cases?